### ISSUE TITLE
[BUG] trap 'exit 0' SIGINT doesn't work in npm scripts

### ISSUE THREAD

**Issue by rafgraph** [npm/cli#1947]
### Current Behavior:
`trap 'exit 0' SIGINT` in an npm script doesn't seem work. I use this to preform clean up after exiting running `watch` scripts with `ctrl c` (maybe there's a better way to do this?).
```
  "scripts": {
    "dev": "npm link && npm run watch && npm unlink -g react-router-hash-link",
    "watch": "trap 'exit 0' SIGINT; babel src -d lib --watch --presets=@babel/preset-env,@babel/preset-react"
  },
```
In npm v6, `npm run dev` links the package to global `node_modules`, then runs the watch script which watches for changes and re-compiles, when I'm done I `ctrl c`, the watch script stops and `npm unlink ...` is run to preform cleanup.

In npm v7, after I `ctrl c`, that's it, nothing runs. I ran it with `--loglevel silly` but nothing is printed to the terminal after I `ctrl c`.

### Expected Behavior:
For the `watch` script to exit with 0 after I press `ctrl c` and `npm unlink ...` to run.

### Steps To Reproduce:
You can reproduce it using my [`react-router-hash-link`](https://github.com/rafgraph/react-router-hash-link) repo (the dev setup is pretty simple).
```shell
git clone git@github.com:rafgraph/react-router-hash-link.git
cd react-router-hash-link
npm install
npm run dev
# the watch script should be running...
# press ctrl c
```

### Environment:
- OS: MacOS 10.15.7
- Node: 12.8.3
- npm: 7.0.0-rc.4

**Comment by nlf** [comment]
This is interesting.. I'll do a little research and see if I can figure out what might've broken this.

~In the mean time, a workaround that should do what you want would be to make your `watch` script look like:~

```
"watch": "babel src -d lib --watch --presets=@babel/preset-env,@babel/preset-react || true"
```

Leaving the suggestion, but after some manual testing the proposed workaround does _not_ work. I'll keep looking!

**Comment by rafgraph** [comment]
@nlf any update on this? I'm still having this issue in v7.0.3 (it's preventing me from using npm v7 regularly). Thanks.

**Comment by nlf** [comment]
i've identified the issue and am working on a fix, hopefully we'll have something for you soon!

**Comment by nlf** [comment]
hi @rafgraph! we published 7.0.6 of npm today that should resolve this for you, can you update with `npm install -g npm@7` and let me know how it goes?

**Comment by rafgraph** [comment]
Yup, works great, thanks @nlf
